### PR TITLE
Changing the names of PnP type

### PR DESCRIPTION
**Pull request by abimishr**
Problem: API and Playbook were having different conventions for PnP type. Now we have changed it.
Fix: Have aligned it with the convention of API. Along with that have fixed for reset.

### CODE REVIEW

File: plugins/modules/pnp_intent.py [hunk 1]
```diff
@@ -62,7 +62,10 @@
         type: str
         default: Onboarding Configuration
       pnp_type:
-        description: Device type of the Pnp device (Default/catalyst_wlc/access_point/stack_switch)
+        description: Device type of the Pnp device (Default/CatalystWLC/AccessPoint/StackSwitch). Default can be
```
Comment (madhansansel) [inline plugins/modules/pnp_intent.py:65]: Can we write as points for more clarity? And StackSwitch there is a spelling error "acting" ..
For example..

```
pnp_type:
  description: Specifies the device type for the Plug and Play (PnP) device.
    - Options include 'Default', 'CatalystWLC', 'AccessPoint', or 'StackSwitch'.
    - 'Default' is applicable to switches and routers.
    - 'CatalystWLC' should be selected for 9800 series wireless controllers.
    - 'AccessPoint' is used when claiming an access point.
    - 'StackSwitch' should be chosen for a group of switches that operate as a single switch, typically used in the access layer.
  type: str
  choices: [ 'Default', 'CatalystWLC', 'AccessPoint', 'StackSwitch' ]
  default: 'Default'
```

File: playbooks/PnP_Workflow_Manager_Playbook.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
+- name: Manage operations - Add, claim, and delete devices of Onboarding Configuration (PnP)
```
Comment (madhansansel) [inline playbooks/PnP_Workflow_Manager_Playbook.yml:2]: Can we write task name with description?
```
- name: Manage PnP Operations
  description:
    - Handles the addition, claiming, and deletion of devices within the Plug and Play (PnP) Onboarding Configuration.
```

File: playbooks/PnP_Workflow_Manager_Playbook.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        config:
+          - site_name: Global/USA/San Francisco/BGL_18
+            device_info:
+              - serial_number: FJC2330E0BB
+                hostname: Test-9300-10
+                state: Unclaimed
+                pid: c9300-24P
+                is_sudi_required: True
+
+    - name: Claim an added Switch with template and image upgrade to a site only
```
Comment (madhansansel) [inline playbooks/PnP_Workflow_Manager_Playbook.yml:59]: - name: Claim a pre-added switch, apply a template, and perform an image upgrade for a specific site

File: playbooks/PnP_Workflow_Manager_Playbook.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            template_details:
+              hostname: SJC-Switch-1
+              interface: TwoGigabitEthernet1/0/2
+            device_info:
+              - serial_number: FJC271924EQ
+                hostname: Switch
+                state: Unclaimed
+                pid: C9300-48UXM
+
+    - name: Claim an added Wireless Controller with template and image upgrade to a site only
```
Comment (madhansansel) [inline playbooks/PnP_Workflow_Manager_Playbook.yml:78]: - name: Claim an existing Wireless Controller, apply a template, and upgrade its image for a specified site

File: playbooks/PnP_Workflow_Manager_Playbook.yml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                hostname: New_WLC
+                state: Unclaimed
+                pid: C9800-CL-K9
+            gateway: 204.192.101.1
+            ip_interface_name: TenGigabitEthernet0/0/0
+            static_ip: 204.192.101.10
+            subnet_mask: 255.255.255.0
+            vlan_id: 1101
+
+    - name: Delete multiple devices from the Pnp dashboard #If device is not present it won't fail
```
Comment (madhansansel) [inline playbooks/PnP_Workflow_Manager_Playbook.yml:101]: - name: Remove multiple devices from the PnP dashboard safely (ignores non-existent devices)

File: plugins/modules/pnp_intent.py [hunk 6]
```diff
@@ -78,7 +81,8 @@
         description: Vlan Id allocated for claimimg of Wireless Controller
         type: str
       ip_interface_name:
-        description: Name of the Interface used for Pnp by the Wireless Controller
+        description: Name of the Interface used for Pnp by the Wireless Controller. It should be configured on the Controller
```
Comment (madhansansel) [inline plugins/modules/pnp_intent.py:84]: description: Specifies the interface name utilized for Plug and Play (PnP) by the Wireless Controller. Ensure this interface is pre-configured on the Controller prior to device claiming.

File: plugins/modules/pnp_intent.py [hunk 7]
```diff
@@ -78,7 +81,8 @@
...
+            before claiming.
         type: str
       rf_profile:
         description: Radio frequecy profile of the AP being claimed (HIGH/LOW/TYPICAL)
```
Comment (madhansansel) [inline plugins/modules/pnp_intent.py:88]: What is HIGH, LOW, TYPICAL?

For device info we can enhance like below.
```
 description:
    - Provides the device-specific information required for adding devices to the PnP database that are not already present.
    - For adding a single device, the list should contain exactly one set of device information. If a site name is also provided, the device can be claimed immediately after being added.
    - For bulk import, the list must contain information for more than one device. Bulk import is intended solely for adding devices; claiming must be performed with separate tasks or configurations.
  type: list
```

```
hostname:
  description:
    - Defines the desired hostname for the PnP device after it has been claimed.
    - The hostname can only be assigned or changed during the claim process, not during bulk or single device additions.

```

What is meant by 'Unclaimed', 'Claimed', or 'Provisioned'? Can you add more information?
```
state:
  description:
    - Represents the onboarding state of the PnP device.
    - Possible values are 'Unclaimed', 'Claimed', or 'Provisioned'.
```

File: plugins/modules/pnp_intent.py (hunk 1 above)
Comment (abimishr) [reply plugins/modules/pnp_intent.py:65]: Done

File: playbooks/PnP_Workflow_Manager_Playbook.yml (hunk 3 above)
Comment (abimishr) [reply playbooks/PnP_Workflow_Manager_Playbook.yml:59]: Done

File: playbooks/PnP_Workflow_Manager_Playbook.yml (hunk 4 above)
Comment (abimishr) [reply playbooks/PnP_Workflow_Manager_Playbook.yml:78]: Done

File: playbooks/PnP_Workflow_Manager_Playbook.yml (hunk 5 above)
Comment (abimishr) [reply playbooks/PnP_Workflow_Manager_Playbook.yml:101]: Done

File: plugins/modules/pnp_intent.py (hunk 6 above)
Comment (abimishr) [reply plugins/modules/pnp_intent.py:84]: Done

File: plugins/modules/pnp_intent.py (hunk 7 above)
Comment (abimishr) [reply plugins/modules/pnp_intent.py:88]: Done

**Review by madhansansel — APPROVED** [review]
Please address all the comments in next PR.

**Pull request merged**